test/prospectortest.py

```diff
@@ -48,23 +50,24 @@ def test_prospectorfail(self):
         base_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'prospectortest')
         test_files = glob.glob(os.path.join(base_dir, 'lib', 'vsc', 'mockinstall', "*.py"))
         test_files = [x for x in test_files if '__init__.py' not in x]
-        log.debug("test_files = %s" % test_files)
-        log.debug("base_dir = %s" % base_dir)
+        log.debug("test_files = %s", test_files)
+        log.debug("base_dir = %s", base_dir)
 
         failures = commontest.run_prospector(base_dir, clear_ignore_patterns=True)
-        log.debug("Failures = %s" % failures)
+        log.debug("Failures = %s", failures)
 
         detected_tests = []
         all_tests = []
         for testfile in test_files:
             testfile_base = os.path.splitext(os.path.basename(testfile))[0].replace("_", "-")
             all_tests.append(testfile_base)
             for failure in failures:
-                if failure['location']['path'] == testfile and testfile_base in [failure['code'], failure['message']]:
+                # old prospector returns strings, new prospector returns Path
+                if Path(failure['location']['path']) == Path(testfile) or testfile_base in [failure['code'], failure['message']]:
```

Comment:
line too long?

Reply:
it's in the test directory, so does not get checked

Reply:
fixed